src/common/lib/client/rest.ts

```diff
@@ -210,31 +210,23 @@ export class Rest {
     if (this.client.options.headers) Utils.mixin(headers, this.client.options.headers);
 
     const requestBody = Utils.encodeBody(requestBodyDTO, this.client._MsgPack, format);
-    Resource.post(
-      this.client,
-      '/messages',
-      requestBody,
-      headers,
-      { newBatchResponse: 'true' },
-      null,
-      (err, body, headers, unpacked) => {
-        if (err) {
-          callback(err);
-          return;
-        }
+    Resource.post(this.client, '/messages', requestBody, headers, {}, null, (err, body, headers, unpacked) => {
+      if (err) {
+        callback(err);
+        return;
+      }
 
-        const batchResults = (
-          unpacked ? body : Utils.decodeBody(body, this.client._MsgPack, format)
-        ) as BatchPublishResult[];
+      const batchResults = (
+        unpacked ? body : Utils.decodeBody(body, this.client._MsgPack, format)
+      ) as BatchPublishResult[];
 
-        // I don't love the below type assertions for `callback` but not sure how to avoid them
-        if (singleSpecMode) {
-          (callback as StandardCallback<BatchPublishResult>)(null, batchResults[0]);
-        } else {
-          (callback as StandardCallback<BatchPublishResult[]>)(null, batchResults);
-        }
+      // I don't love the below type assertions for `callback` but not sure how to avoid them
```

Comment:
For internal typings I don't think we would really lose any strictness by just doing:
```ts
  batchPublish<T extends BatchPublishSpec | BatchPublishSpec[]>(
    specOrSpecs: T,
    callback?: StandardCallback<BatchPublishResult | BatchPublishResult[]>
  ): void | Promise<BatchPublishResult | BatchPublishResult[]> {
```

If we did that, there would be no need to use type assertions here

Reply:
Fair point; on the other hand I guess it's nice to be able to have an overload that matches the type declared in `ably.d.ts`.